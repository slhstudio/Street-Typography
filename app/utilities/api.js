import axios from 'axios';

const api = {}

api.isSignedIn = async (error) => {
  const result = await axios.get('/user')
    .catch(error);
  return result.data;
}

module.exports = api;