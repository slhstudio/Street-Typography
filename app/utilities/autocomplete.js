function autocomplete(input, latInput, lngInput) {
  if (!input) return;
  console.log(input);
  const dropdown = new google.maps.places.Autocomplete(input);
}

export default autocomplete;