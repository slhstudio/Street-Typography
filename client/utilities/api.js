import axios from 'axios';

export async function isSignedIn (error) {
  const result = await axios.get('/user')
    .catch(error);
  return result.data;
}

export async function findAllPhotos (error) {
  const result = await axios.get('/findAllPhotos')
    .catch(error);
  return result;
}

