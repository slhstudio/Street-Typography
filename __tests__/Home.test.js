import React from 'react';
import HomeComponent from '../client/components/Home';
import { shallow, mount } from 'enzyme';
import { findAllPhotos as mockFindAllPhotos, isSignedIn as mockIsSignedIn} from '../client/utilities/api'
import toJSON from 'enzyme-to-json';



jest.mock('../client/utilities/api', () => {
  const allPhotos = [{photo: 'photo1'}, {photo: 'photo2'}]
  const user = 'Sarah';
  return {
    isSignedIn: jest.fn(() => Promise.resolve(user)),
    findAllPhotos: jest.fn(() => Promise.resolve({data: allPhotos }))
  }
})

const wait = (amount = 0) => new Promise(resolve => setTimeout(resolve, amount));

describe('<Home/>', () => {
  it ('renders and displays properly', () => {
    const wrapper = shallow(<HomeComponent isUser={true}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
    wrapper.setState({ loading : false});
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  it('finds all photos', async () => {
    expect.assertions(2);
    const data = await mockFindAllPhotos();
    expect(data).toEqual({data: [{photo: 'photo1'}, {photo: 'photo2'}]})
    expect(mockFindAllPhotos).toHaveBeenCalledTimes(2);
    // expect(mockIsSignedIn).toHaveBeenCalledTimes(1);
  });
  it('gets user', async () => {
    expect.assertions(1);
    const user = await mockIsSignedIn();
    expect(user).toBe('Sarah');
  });
  it('should call function passed in as prop with correct argument', async () => {
    const mockProps = {
      isUser: false,
      handleLogIn : jest.fn()
    }
    const wrapper = shallow(<HomeComponent {...mockProps} />);
    wrapper.instance().componentDidMount();
    await wait();
    expect(mockProps.handleLogIn).toHaveBeenCalledTimes(2);
    expect(mockProps.handleLogIn).toHaveBeenCalledWith('Sarah');
  })
});


