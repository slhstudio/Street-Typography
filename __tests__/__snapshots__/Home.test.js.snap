// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home/> renders and displays properly 1`] = `
<div>
  <Loading
    speed={300}
    text="LOADING"
  />
</div>
`;

exports[`<Home/> renders and displays properly 2`] = `
<div>
  <PhotoGrid
    notes={Array []}
    photos={Array []}
  />
</div>
`;
